### DISCUSSION TITLE
Is it possible that vim9 release is disfunctional with coc.nvim?

### DISCUSSION THREAD

**Discussion by rshutt-va** [neoclide/coc.nvim#3905] [Q&A] [answered]
Is it possible that the vim9 release is dysfunctional with coc.nvim?

Homebrew released vim9 2 days ago, and since then, whenever I have a file like a .json document or a .tf HCL document loaded and when I am in Insert or Append mode, I have an issue with inserting a newline.  When I press RETURN to start a new line, the CPU will go to 100% on a single core, and VIM will appear locked up until I hit `C-c`.  If I remove `coc.nvim`, this no longer happens.  

I am just wondering.  I've not had a lot of time to debug.

**Comment by fannheyward** [comment, marked as answer]
What's your coc.nvim `:CocInfo`?

Tested with coc.nvim master/release on Vim 9.0, works ok

```
vim version: VIM - Vi IMproved 9.0 9000000¬
node version: v16.15.1¬
coc.nvim version: 0.0.81-2522eee5 2022-06-14 19:03:29 +0800¬
coc.nvim directory: /Users/fannheyward/.local/share/nvim/plugged/coc.nvim
```

**Comment by rshutt-va** [comment]
Definitely not master :). Running from `release` since day 1 of using coc.nvim.

I guess I should give that a shot.

```
## versions

vim version: VIM - Vi IMproved 9.0 9000000
node version: v18.4.0
coc.nvim version: 0.0.81-2522eee5 2022-06-14 19:03:29 +0800
coc.nvim directory: /Users/randall/.vim/pack/randallman/start/coc.nvim
term: tmux
platform: darwin

## Log of coc.nvim

2022-07-06T10:33:13.743 INFO (pid:5127) [plugin] - coc.nvim initialized with node: v18.4.0 after 58ms
2022-07-06T10:33:15.095 INFO (pid:5127) [attach] - receive notification: checkJsonExtension []
2022-07-06T10:33:15.125 INFO (pid:5127) [services] - registered service "json"
2022-07-06T10:33:15.125 INFO (pid:5127) [services] - Json language server state change: stopped => starting
2022-07-06T10:33:15.212 INFO (pid:5127) [services] - Json language server state change: starting => running
2022-07-06T10:33:15.216 INFO (pid:5127) [services] - service json started
2022-07-06T10:33:17.605 INFO (pid:5127) [attach] - receive notification: showInfo []
```

**Reply by rshutt-va** [reply]
Ok bumped up to master and the problem seems to be gone!

Looks like a very large number of commits between master and release!

**Reply by fannheyward** [reply]
You don't need to care about the number of commits between master and release. The `release` branch was built on https://github.com/neoclide/coc.nvim/commit/2522eee57d18aea040f4def29f2b8e9c69b5690c.

**Comment by rshutt-va** [comment]
Actually, I have to retract my statement about the problem being gone.  It's just manifesting differently.  I'm not able to identify the specific causes now, but if the conditions are right, attempting to end a line and start a new line will cause the CPU to peg to 99/10-0 (1 core) and I have to C-c to regain control of the vim9.0.0 process.  Ill have to put some more time into debugging sometime in the near future.

**Comment by chemzqm** [comment]
To see what function block your vim, checkout `:h :profile`.
Could be something get slow during `coc#on_enter()`, try avoid it to see if it helps.

**Reply by rshutt-va** [reply]
I'll check it out!  I do miss coc-nvim in my toolset of late...
